plugins/vpc-shared-eni/network/bridge_windows.go

```diff
@@ -132,31 +145,29 @@ func (nb *BridgeBuilder) FindOrCreateEndpoint(nw *Network, ep *Endpoint) error {
 		return fmt.Errorf("Only a single IPv4 address per endpoint is supported on Windows")
 	}
 
-	// Query the infrastructure container ID.
-	isInfraContainer, infraContainerID, err := nb.getInfraContainerID(ep)
-	if err != nil {
-		return err
-	}
+	// Query the namespace identifier.
+	nsType, namespaceIdentifier := nb.getNamespaceIdentifier(ep)
 
 	// Check if the endpoint already exists.
-	endpointName := nb.generateHNSEndpointName(ep, infraContainerID)
+	endpointName := nb.generateHNSEndpointName(ep, namespaceIdentifier)
 	hnsEndpoint, err := hcsshim.GetHNSEndpointByName(endpointName)
 	if err == nil {
 		log.Infof("Found existing HNS endpoint %s.", endpointName)
-		if isInfraContainer {
+		if nsType == infraContainerNS || nsType == hcsNamespace {
 			// This is a benign duplicate create call for an existing endpoint.
 			// The endpoint was already attached in a previous call. Ignore and return success.
 			log.Infof("HNS endpoint %s is already attached to container ID %s.",
 				endpointName, ep.ContainerID)
 		} else {
 			// Attach the existing endpoint to the container's network namespace.
-			err = nb.attachEndpoint(hnsEndpoint, ep.ContainerID)
+			// Attachment of endpoint to each container would occur only when using HNS V1 APIs.
+			err = nb.attachEndpointV1(hnsEndpoint, ep.ContainerID)
```

Comment:
Confused on this case. Wouldnt this be a hcn v2 call if we are using hcn namespaces?

Reply:
Oh sorry read the `if` wrong. This case is for the nonInfraContainer. This is ok